src/drivers/utils/opfs-utils.ts

```diff
@@ -0,0 +1,394 @@
+  if (!normalizedPath.includes("/")) return "";
+  return normalizedPath.slice(0, normalizedPath.lastIndexOf("/"));
+}
+
+/**
+ * These correspond with the handle's `kind` property
+ */
+type SpecificHandleType = "file" | "directory";
+type UnspecificHandleType = SpecificHandleType | "any";
+
+/**
+ * Options for the `getHandle` function
+ */
+type GetHandleOptions = {
+  /**
+   * Whether to create the handle if it doesn't exist
+   * Setting to `true` requires setting a specific `type`
+   *
+   * @default false
+   */
+  create?: boolean;
+
+  /**
+   * Which kind of handle is desired ('file', 'directory', or 'any')
+   * If a handle of the provided type cannot be obtained, an error will be thrown
+   *
+   * @default "any"
+   */
+  type?: UnspecificHandleType;
+};
+
+/**
+ * This is a special subset of the `GetHandleOptions` type for improved
+ * type inference when overloading the `getHandle` function
+ *
+ * It represents options with the `create` option set to `true`,
+ * which requires setting a known `type`
+ */
+type GetHandleWithCreateOptions<T extends SpecificHandleType> = {
+  create: true;
+  type: T;
+};
+
+/**
+ * This is a special subset of the `GetHandleOptions` type for improved
+ * type inference when overloading the `getHandle` function
+ *
+ * It represents options with the `create` option set to false or missing,
+ * which allows for a less specific `type` option
+ */
+type GetHandleWithoutCreateOptions<T extends UnspecificHandleType> = {
+  create?: false;
+  type?: T;
+};
+
+type HandleTypesByName = {
+  file: FileSystemFileHandle;
+  directory: FileSystemDirectoryHandle;
+  any: FileSystemHandle;
+};
+
+async function getHandle<T extends SpecificHandleType>(
+  root: FileSystemDirectoryHandle,
+  path: string | string[],
+  opts: GetHandleWithCreateOptions<T>
+): Promise<HandleTypesByName[T]>;
+async function getHandle<T extends UnspecificHandleType>(
+  root: FileSystemDirectoryHandle,
+  path: string | string[],
+  opts?: GetHandleWithoutCreateOptions<T>
+): Promise<HandleTypesByName[T]>;
+
+/**
+ * Get a FileSystemHandle for a given path
+ *
+ * @param root The directory handle to start at
+ * @param path The path to the desired handle
+ * @param opts Options for obtaining the handle
+ */
+async function getHandle(
+  root: FileSystemDirectoryHandle,
+  path: string | string[],
+  opts: GetHandleOptions = {}
+): Promise<FileSystemHandle> {
+  // Split the path
+  const segments = Array.isArray(path) ? path : normalizePath(path).split("/");
+
+  // If there are no segments, return the root handle
+  if (segments.length === 0) return root;
+
+  const create = Boolean(opts.create);
+  const handleType: UnspecificHandleType = opts.type ?? "any";
+
+  // If possibly creating the handle, a specific type must be provided
+  if (create && handleType !== "file" && handleType !== "directory") {
+    throw createError(
+      DRIVER_NAME,
+      'Invalid handle type, must be "file" or "directory" when creating'
```

Comment:
Tip: Using backticks, makes better console formatting when consola is enabled. Feel free to update other messages if i missed.

```suggestion
      'Invalid handle type, must be `file` or `directory` when creating'
```

Reply:
Seems like this is the only occurrence, resolved.